scripts/linters/js_ts_linter.py

```diff
@@ -220,6 +228,57 @@ def _compile_ts_file(self, filepath, dir_path):
             dir_path, os.path.basename(filepath).replace('.ts', '.js'))
         return compiled_js_filepath
 
+    def _check_any_type(self):
+        """Checks if the type of any variable is declared as any
+        in typescript files.
+        """
+
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting any type check')
+            python_utils.PRINT('----------------------------------------')
+
+        files_to_check = self.all_filepaths
+        patterns_to_match = [r':\ *any', r'^\ *any']
+        stdout = sys.stdout
+
+        with linter_utils.redirect_stdout(stdout):
+            failed = False
+            summary_messages = []
+
+            for file_path in files_to_check:
+                if file_path in EXCLUDED_FILES_ANY_TYPE_CHECK:
+                    continue
+
+                file_content = FILE_CACHE.read(file_path)
+                starts_with_type = False
+
+                for line_number, line in enumerate(file_content.split('\n')):
+                    if ((starts_with_type and re.findall(
+                            patterns_to_match[1], line)) or re.findall(
+                                patterns_to_match[0], line)):
+                        failed = True
+                        python_utils.PRINT(
+                            '%s --> ANY type found in this file. Line no.'
+                            ' %s' % (
+                                file_path, line_number + 1))
+                        python_utils.PRINT('')
+
+                    if line:
+                        starts_with_type = line[len(line) - 1] == ':'
```

Comment:
What does this exactly do? Maybe add some explanation comments.

Reply:
`starts_with_type` is a variable that tells if the previous line ending with a `:`. So we know this line starts with a type.